src/dakota.py

```diff
@@ -32,25 +32,48 @@
 """
 
 from __future__ import with_statement
+from __future__ import print_function
 
 import logging
 import os
 import pyDAKOTA
+import weakref
+
+# This will hold a reference to the DakotaDriver instance or
+# the user specified custom model instance
+_USER_DATA = weakref.WeakValueDictionary()
+
 
 class DakotaBase(object):
     """ Base class for a DAKOTA 'driver'. """
 
     def __init__(self):
         self.input = DakotaInput()
 
-    def run_dakota(self, infile='dakota.in',
-                   stdout=None, stderr=None, restart=0, other_model=None):
+    def run_dakota(self, infile='dakota.in', stdout=None, stderr=None, restart=0,
+                   self_driver=True, other_model_instance=None):
         """
-        Write `self.input` to `infile`, providing `self` as `data`.
+        Write `self.input` to `infile`, providing `self` as `driver_instance` if requested.
+        If additional model instance is provided pass that as 'driver_instance'
+        If both are specified issue an warning but continue with self
         Then run DAKOTA with that input, MPI specification, and files.
         DAKOTA will then call our :meth:`dakota_callback` during the run.
         """
-        self.input.write_input(infile, data=self, other_data=None)
+        if self_driver and other_model_instance is not None:
+            print("Warning: Both dakota driver and other model instance are provided. "
+                  "Dakota will continue with self as driver")
+
+        # If user specified another driver instance than use that as driver
+        if self_driver:
+            self.input.write_input(infile, driver_instance=self)
+
+        elif other_model_instance is not None:
```

Comment:
I agree with this very much... it was in the original code and I though it might be useful at some point, but indeed I would never use it myself. Seeing you are thinking the same I will remove this possibility and only allow for an instance derived from DakotaBase to be used.